src/game/UICmdList.cpp

```diff
@@ -0,0 +1,283 @@
+    m_bClosingNow = true;
+}
+
+void CUICmdListDlg::Tick() {
+    if (m_bOpenningNow) {
+        POINT ptCur = this->GetPos();
+        RECT  rc = this->GetRegion();
+        float fWidth = (float)(rc.right - rc.left);
+
+        float fDelta = 5000.0f * CN3Base::s_fSecPerFrm;
+        fDelta *= (fWidth - m_fMoveDelta) / fWidth;
+        if (fDelta < 2.0f) {
+            fDelta = 2.0f;
+        }
+        m_fMoveDelta += fDelta;
+
+        int iXLimit = CN3Base::s_CameraData.vp.Width - (int)fWidth;
+        ptCur.x = CN3Base::s_CameraData.vp.Width - (int)m_fMoveDelta;
+        if (ptCur.x <= iXLimit) {
+            ptCur.x = iXLimit;
+            m_bOpenningNow = false;
+        }
+
+        this->SetPos(ptCur.x, ptCur.y);
+    } else if (m_bClosingNow) {
+        POINT ptCur = this->GetPos();
+        RECT  rc = this->GetRegion();
+        float fWidth = (float)(rc.right - rc.left);
+
+        float fDelta = 5000.0f * CN3Base::s_fSecPerFrm;
+        fDelta *= (fWidth - m_fMoveDelta) / fWidth;
+        if (fDelta < 2.0f) {
+            fDelta = 2.0f;
+        }
+        m_fMoveDelta += fDelta;
+
+        int iXLimit = CN3Base::s_CameraData.vp.Width;
+        ptCur.x = CN3Base::s_CameraData.vp.Width - (int)(fWidth - m_fMoveDelta);
+        if (ptCur.x >= iXLimit) {
+            ptCur.x = iXLimit;
+            m_bClosingNow = false;
+
+            this->SetVisibleWithNoSound(false, false, true);
+        }
+
+        this->SetPos(ptCur.x, ptCur.y);
+    }
+
+    CN3UIBase::Tick();
+}
+
+bool CUICmdListDlg::CreateCategoryList() {
+    if (m_pList_CmdCat == NULL || m_pList_Cmds == NULL) {
+        return false;
+    }
+
+    std::string szCategory;
+    int         idStart = IDS_PRIVATE_CMD_CAT;
+
+    for (int i = 0; i < 8; i++) {
+        ::_LoadStringFromResource(i + 7800, szCategory); //load command categories
+
+        if (CGameBase::s_pPlayer->m_InfoBase.iAuthority == 0 && szCategory == "GM") { // if not gm hide this category
+            continue;
+        }
+
+        m_pList_CmdCat->AddString(szCategory);
+        idStart++;
+    }
+
+    m_pList_CmdCat->SetFontColor(0xffffff00); //green
+
+    int idCur = 8000; //Command list strings start at this index
+    int idEnd = 9600; //Command list strings end at this index
+
+    std::string szCommand;
+    //create map of commands
+    for (int i = idCur; idCur < idEnd; idCur++, i++) {
+        if (idCur == 9000) {
+            i += 400; // offset and put gm cmds at end of map
+        } else if (idCur == 9100) {
+            i -= 500;
+            idCur = 9200;
+        }
+        szCommand.clear();
+        ::_LoadStringFromResource(idCur, szCommand);
+        if (!szCommand.empty() && (i / 100) % 2 == 0) {
+            m_mapCmds[i] = szCommand;
+        }
+    }
+
+    UpdateCommandList(CMD_LIST_PRIVATE); //initialize a cmd list for viewing when opening cmd window
+
+    return true;
+}
+bool CUICmdListDlg::UpdateCommandList(uint8_t cmdCat) {
+    if (m_pList_Cmds == NULL) {
```

Comment:
Again, I don't know what we're working with, but I assume we should be moving away from the old `NULL` definitions and explicitly using C++11's `nullptr` instead. But @stevewgr can confirm.

Reply:
Definitely: https://github.com/ko4life-net/ko/pull/222#discussion_r1606102038